orttraining/orttraining/test/training_api/trainer/trainer.cc

```diff
@@ -355,8 +355,14 @@ int RunTraining(const TestRunnerParameters& params) {
         PathString ckpt_file = ConcatPathComponent<PathChar>(params.output_dir, ToPathString(oss.str()));
         ORT_RETURN_ON_ERROR(g_ort_training_api->SaveCheckpoint(ckpt_file.c_str(), session, true));
 
-        // TODO(baiju): enable adding more properties to checkpoint
-        // state_to_save.property_bag.AddProperty<int64_t>(std::string("epoch"), epoch);
+        ORT_RETURN_ON_ERROR(g_ort_training_api->SetCheckpointProperty(
+            checkpoint_state, "epoch", OrtCheckpointPropertyType::IntProperty, &epoch));
```

Comment:
where did checkpoint_state comes from? we should set those property before call SaveCheckpoint, right?

Reply:
Initially, I was thinking we could use the same `CheckpointState` as the user had when they called `LoadCheckpoint`. This would contain all the model parameters, but not the optimizer parameters, so we have a gap.
Then, I was going off what we had originally thought of for checkpoint saving. The two step approach.

1. User calls `GetStateDict` or `GetCheckpoint` on the `TrainingSession`. And we populate the checkpoint state for the user. This has not been implemented yet.
2. User then calls `SetCheckpointProperty` on this `CheckpointState` and following this, calls `SaveCheckpoint`. We will need to update our `SaveCheckpoint` function to support this. 

Let's sync and discuss this further.